### DISCUSSION TITLE
How to list each of upload items progress in a CUSTOM notify ?

### DISCUSSION THREAD

**Discussion by kevinstyle** [Studio-42/elFinder#3307] [Ideas]
Hi, elfinder team

I'd like to dev a custom notify widget that can display each item's Name, Upload Progress, and can Cancel each item by user. Like google drive's behavior and UI, show as below.

![google_drive_like](https://cloud.githubusercontent.com/assets/8855408/18743821/74b89a46-80ed-11e6-9df9-c2959d8d2e8e.png)

I have known that upload behavior is controlled by upload.js and elfinder.js, but it is very difficult to add new notify widget to support this.

Can you team members give me some hint ?

thanks.

**Comment by nao-pon** [comment]
@kevinstyle I also think this is a good idea, but its implementation is very difficult. We might need to change the specification. We are looking for ideas from everyone.

**Comment by CodeLyokoXtEAM** [comment]
It is possible to upload files, one by one separate request in elfinder?

**Comment by nao-pon** [comment]
> It is possible to upload files, one by one separate request in elfinder?

Yes, it is possible by result `$results['uplMaxFile'] = 1` on initialization of the open command.

example for test (PHP >= 5.4)

``` php
$opts = [
    'bind' => [
        'open' => [
            function($cmd, &$result, $args) {
                if (! empty($args['init'])) {
                    $result['uplMaxFile'] = 1;
                }
            }
        ]
    ],
    'roots' => [
        [
            'driver' => 'LocalFileSystem',
            'path'   => '/path/to/files/',
            'URL'    => 'http://localhost/to/files/'
        ]
    ]
];
```

**Comment by diego3** [comment]
+1, Nice Widget
